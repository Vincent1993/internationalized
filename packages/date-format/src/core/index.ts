export { DateFormatProvider, useDateFormatContext } from './context';
export { createDateFormatter, resolveDateFormatOptions } from './formatter';
export type {
  CreateDateFormatOptions,
  DateFormatContextValue,
  DateFormatter,
  DateFormatProviderProps,
  ResolvedDateFormatOptions,
  UseDateFormatOptions,
  UseDateFormatResult,
  DateInput,
} from './types';
