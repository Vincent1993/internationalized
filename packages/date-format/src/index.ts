export { DateFormat } from './components';
export type { DateFormatProps } from './components';

export { useDateFormat } from './hooks';
export type { UseDateFormatOptions, DateFormatter, UseDateFormatResult } from './hooks';

export {
  DateFormatProvider,
  useDateFormatContext,
  createDateFormatter,
  resolveDateFormatOptions,
} from './core';
export type {
  DateFormatContextValue,
  DateFormatProviderProps,
  CreateDateFormatOptions,
  ResolvedDateFormatOptions,
  DateInput,
} from './core';
