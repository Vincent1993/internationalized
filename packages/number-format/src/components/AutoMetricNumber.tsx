import { forwardRef, ElementType, memo } from 'react';
import type { ReactNode } from 'react';
import { useAutoFormat } from '../hooks/useAutoFormat';
import type { MetricFormatRule, UseFormatResult } from '../core/types';
import { BaseNumberFormat } from './_internal/BaseNumberFormat';

export type AutoMetricNumberProps<T extends ElementType = 'span'> = Omit<
  React.ComponentPropsWithoutRef<T>,
  'children'
> & {
  value: unknown;
  name: string;
  rules?: MetricFormatRule[];
  fallback?: ReactNode;
  asChild?: boolean;
  as?: T;
  children?: ReactNode | ((result: UseFormatResult) => ReactNode);
};

/**
 * @name AutoMetricNumber
 * @description 一个智能的指标格式化组件，可根据指标名称自动应用预设的格式化规则。
 *   - 核心属性：`value` 数值、`name` 指标名称、`rules` 规则配置
 *   - 渲染属性：`as` 多态元素、`asChild` 插槽模式、`fallback` 错误回退
 *   - 交互属性：`children` 函数式渲染（接收 `UseFormatResult`）
 *   - 支持所有 React 元素属性（继承自 `React.ComponentPropsWithoutRef<T>`）
 *   - 规则可以由 `NumberFormatProvider` 全局提供，也可以由组件局部提供
 *   - 优先级：局部 `rules` > 全局 `rules` > 默认配置
 *
 * @param value - 要格式化的数值
 * @param name - 指标名称，用于匹配格式化规则
 * @param rules - 局部格式化规则数组
 * @param fallback - 错误时的回退内容
 * @param asChild - 是否作为子组件渲染（插槽模式）
 * @param as - 渲染的 HTML 元素类型，默认为 'span'
 * @param children - 函数式子组件，接收 `UseFormatResult` 参数
 * @param ...props - 所有其他 React 元素属性
 *
 * @since 0.0.1
 *
 * @example
 * ```tsx
 * // 1. 在 Provider 中定义全局规则
 * const globalRules = [
 *   { name: 'revenue', options: { style: 'currency', currency: 'USD' } },
 *   { pattern: /rate$/, options: { style: 'percent', maximumFractionDigits: 2 } },
 * ];
 *
 * <NumberFormatProvider options={{ rules: globalRules }}>
 *   <Dashboard />
 * </NumberFormatProvider>
 *
 * // 2. 在 Dashboard 组件中使用
 * function Dashboard() {
 *   return (
 *     <div>
 *       // 匹配到名为 'revenue' 的精确规则
 *       <AutoMetricNumber name="revenue" value={12345.67} />
 *       // => <span>$12,345.67</span>
 *
 *       // 匹配到 `/_rate$/` 的正则规则
 *       <AutoMetricNumber name="conversion_rate" value={0.12345} />
 *       // => <span>12.35%</span>
 *
 *       // 使用局部 `rules` 覆盖全局规则
 *       <AutoMetricNumber
 *         name="revenue"
 *         value={12345.67}
 *         rules={[{ name: 'revenue', options: { currency: 'EUR' } }]}
 *       />
 *       // => <span>€12,345.67</span>
 *
 *       // 使用函数式 children 自定义渲染
 *       <AutoMetricNumber name="growth_rate" value={-0.05}>
 *         {(result) => (
 *           <strong className={result.sign.isNegative ? 'text-red-500' : 'text-green-500'}>
 *             {result.formattedValue}
 *           </strong>
 *         )}
 *       </AutoMetricNumber>
 *       // => <strong class="text-red-500">-5%</strong>
 *     </div>
 *   );
 * }
 * ```
 */
const AutoMetricNumberImpl = forwardRef(
  <T extends ElementType = 'span'>(
    props: AutoMetricNumberProps<T>,
    ref: React.ComponentPropsWithRef<T>['ref'],
  ) => {
    const { name, rules, ...rest } = props;
    const formatter = useAutoFormat({ name, rules });

    return <BaseNumberFormat {...rest} formatter={formatter} ref={ref} />;
  },
);

AutoMetricNumberImpl.displayName = 'AutoMetricNumber';

export const AutoMetricNumber = memo(AutoMetricNumberImpl) as <T extends ElementType = 'span'>(
  props: AutoMetricNumberProps<T> & { ref?: React.ComponentPropsWithRef<T>['ref'] },
) => JSX.Element;
