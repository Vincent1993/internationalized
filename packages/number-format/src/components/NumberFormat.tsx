import { forwardRef, ElementType, memo } from 'react';
import type { ReactNode } from 'react';
import { useFormat } from '../hooks/useFormat';
import type { UseFormatOptions, UseFormatResult } from '../core/types';
import { BaseNumberFormat } from './_internal/BaseNumberFormat';

export type NumberFormatProps<T extends ElementType = 'span'> = Omit<
  React.ComponentPropsWithoutRef<T>,
  'children'
> & {
  value: unknown;
  options?: UseFormatOptions;
  fallback?: ReactNode;
  asChild?: boolean;
  as?: T;
  children?: ReactNode | ((result: UseFormatResult) => ReactNode);
};

/**
 * @name NumberFormat
 * @description 一个功能强大的、支持多态的数字格式化组件。
 *   - 核心属性：`value` 数值、`options` 格式化配置（继承自 `useFormat` hook）
 *   - 渲染属性：`as` 多态元素、`asChild` 插槽模式、`fallback` 错误回退
 *   - 交互属性：`children` 函数式渲染（接收 `UseFormatResult`）
 *   - 支持所有 React 元素属性（继承自 `React.ComponentPropsWithoutRef<T>`）
 *   - 自动处理 `null`, `undefined`, `NaN` 等无效值
 *
 * @param value - 要格式化的数值
 * @param options - 格式化配置选项，继承自 `useFormat` hook 的 `UseFormatOptions`
 * @param fallback - 错误时的回退内容
 * @param asChild - 是否作为子组件渲染（插槽模式）
 * @param as - 渲染的 HTML 元素类型，默认为 'span'
 * @param children - 函数式子组件，接收 `UseFormatResult` 参数
 * @param ...props - 所有其他 React 元素属性
 *
 * @since 0.0.1
 *
 * @example
 * ```tsx
 * // 基础用法
 * <NumberFormat value={1234.56} options={{ style: "decimal" }} />
 * // => <span>1,234.56</span>
 *
 * // 货币格式
 * <NumberFormat value={1234.56} options={{ style: "currency", currency: "CNY" }} />
 * // => <span>¥1,234.56</span>
 *
 * // 使用 `as` 属性渲染为 <div>
 * <NumberFormat value={99} as="div" className="font-bold" />
 * // => <div class="font-bold">99</div>
 *
 * // 函数式 children，用于自定义渲染
 * <NumberFormat value={-0.05} options={{ style: "percent" }}>
 *   {(result) => (
 *     <span className={result.sign.isNegative ? 'text-red-500' : 'text-green-500'}>
 *       {result.formattedValue}
 *     </span>
 *   )}
 * </NumberFormat>
 * // => <span class="text-red-500">-5%</span>
 *
 * // `asChild` 模式，将内容注入到子按钮
 * <NumberFormat value={42} asChild>
 *   <button>Count: </button>
 * </NumberFormat>
 * // => <button>Count: 42</button>
 * ```
 */
const NumberFormatImpl = forwardRef(
  <T extends ElementType = 'span'>(
    props: NumberFormatProps<T>,
    ref: React.ComponentPropsWithRef<T>['ref'],
  ) => {
    const { options, ...rest } = props;
    const formatter = useFormat(options);

    return <BaseNumberFormat {...rest} formatter={formatter} ref={ref} />;
  },
);

NumberFormatImpl.displayName = 'NumberFormat';

export const NumberFormat = memo(NumberFormatImpl) as <T extends ElementType = 'span'>(
  props: NumberFormatProps<T> & { ref?: React.ComponentPropsWithRef<T>['ref'] },
) => JSX.Element;
