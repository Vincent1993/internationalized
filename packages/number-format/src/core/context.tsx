import { createContext, useContext, useMemo } from 'react';
import type { NumberFormatContextValue, NumberFormatProviderProps } from './types';
import { mergeRules } from './utils/rules';

const NumberFormatContext = createContext<NumberFormatContextValue | null>(null);

/**
 * @name NumberFormatProvider
 * @description 为应用提供全局的数字格式化配置上下文，支持所有 `Intl.NumberFormat`
 * 选项与指标规则合并。
 *
 * @since 0.0.1
 *
 * @example
 * ```tsx
 * import { NumberFormatProvider, NumberFormat, AutoMetricNumber } from '@your-package/number-format';
 *
 * // 推荐用法：使用 options 对象
 * function App() {
 *   return (
 *     <NumberFormatProvider
 *       options={{
 *         locale: "zh-CN",
 *         useGrouping: true,
 *         minimumFractionDigits: 2
 *       }}
 *     >
 *       <Dashboard />
 *     </NumberFormatProvider>
 *   );
 * }
 *
 * // 带指标规则的配置
 * function AppWithRules() {
 *   const options = {
 *     locale: "zh-CN",
 *     useGrouping: true,
 *     rules: [
 *       // 精确匹配规则
 *       { name: 'revenue', options: { style: 'currency', currency: 'CNY' } },
 *       { name: 'profit', options: { style: 'currency', currency: 'CNY' } },
 *       { name: 'cost', options: { style: 'currency', currency: 'CNY' } },
 *
 *       // 正则匹配规则
 *       { pattern: '.*_rate$', options: { style: 'percent', maximumFractionDigits: 2 } },
 *       { pattern: '.*_ratio$', options: { style: 'percent', maximumFractionDigits: 1 } },
 *       { pattern: '.*_count$', options: { style: 'decimal', useGrouping: true } },
 *       { pattern: '.*_permille$', options: { style: 'per-mille' } },
 *       { pattern: '.*_permyriad$', options: { style: 'per-myriad' } },
 *       { pattern: '.*_pp$', options: { style: 'percentage-point' } },
 *       { pattern: '.*_cn_upper$', options: { style: 'cn-upper' } },
 *     ]
 *   };
 *
 *   return (
 *     <NumberFormatProvider options={options}>
 *       <MetricDashboard />
 *     </NumberFormatProvider>
 *   );
 * }
 *
 * // 嵌套 Provider（局部覆盖全局配置）
 * function NestedProviders() {
 *   return (
 *     <NumberFormatProvider options={{ locale: "zh-CN", useGrouping: true }}>
 *       <div>
 *         <h1>Global Config</h1>
 *         <NumberFormat value={1234.56} />
 *
 *         <NumberFormatProvider options={{ locale: "en-US", style: "currency", currency: "USD" }}>
 *           <div>
 *             <h2>Local Override</h2>
 *             <NumberFormat value={1234.56} />
 *           </div>
 *         </NumberFormatProvider>
 *       </div>
 *     </NumberFormatProvider>
 *   );
 * }
 * ```
 */
export const NumberFormatProvider = ({ children, options }: NumberFormatProviderProps) => {
  const parentValue = useNumberFormatContext();

  const value = useMemo<NumberFormatContextValue | null>(() => {
    if (!options) {
      return parentValue;
    }

    if (!parentValue) {
      return options;
    }

    const mergedValue: NumberFormatContextValue = {
      ...parentValue,
      ...options,
      rules: mergeRules(parentValue.rules || [], options.rules || []),
    };

    if (mergedValue.rules?.length === 0) {
      delete mergedValue.rules;
    }

    return mergedValue;
  }, [parentValue, options]);

  return <NumberFormatContext.Provider value={value}>{children}</NumberFormatContext.Provider>;
};

/**
 * @name useNumberFormatContext
 * @description React Hook，用于获取当前的数字格式化上下文配置，主要供内部 Hook
 * 与组件使用，也可用于自定义封装。
 *
 * @since 0.0.1
 *
 * @returns 当前上下文配置；若不存在 Provider 则返回 `null`
 */
export const useNumberFormatContext = (): NumberFormatContextValue | null => {
  return useContext(NumberFormatContext);
};
