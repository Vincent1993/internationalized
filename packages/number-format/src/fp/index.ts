/**
 * @file 函数式编程 (FP) 入口点
 * @description
 *   为常见的数字格式化任务提供纯粹的、无副作用的函数。
 */

// 格式化函数
export {
  formatAsDecimal,
  formatAsInteger,
  formatAsCurrency,
  formatAsPercent,
  formatAsPerMille,
  formatAsPerMyriad,
  formatAsPercentagePoint,
  formatAsChineseUppercase,
  formatAsCompact,
  formatAsScientific,
  formatAsDecimalEx,
  formatAsIntegerEx,
  formatAsCurrencyEx,
  formatAsPercentEx,
  formatAsPerMilleEx,
  formatAsPerMyriadEx,
  formatAsPercentagePointEx,
  formatAsChineseUppercaseEx,
  formatAsCompactEx,
  formatAsScientificEx,
} from './formatters';

// 模板格式化
export { formatWithTemplate, formatWithTemplateEx } from './templates';

// 解析函数
export {
  parseDecimal,
  parseInteger,
  parseCurrency,
  parsePercent,
  parsePerMille,
  parsePerMyriad,
  parsePercentagePoint,
  parseChineseUppercase,
  parseCompact,
  parseScientific,
  parseAuto,
} from './parsers';

// 缓存管理
export { clearAllFPCaches, clearCache, clearFormatterCache, clearParserCache } from './memoize';

// 配置管理
export { config, resetDefaultConfigs, type FormatterDefaults } from './defaults';

// 模板配置
export {
  configTemplates,
  resetTemplateConfig,
  type TemplateConfigState,
  type TemplateConfigEntry,
  type TemplateConfigurationInput,
} from './templates';

// 核心类型
export type { ParseResult, ParseOptions } from '../core/parser';
