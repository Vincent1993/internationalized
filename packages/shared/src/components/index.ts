export { Slot } from './Slot';
export { createBaseFormatComponent } from './createBaseFormatComponent';
export type { BaseFormatComponentProps, FormatInvoker } from './createBaseFormatComponent';
