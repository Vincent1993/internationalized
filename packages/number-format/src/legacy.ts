import { NumberFormat } from './components';
import { NumberFormatProvider } from './core';

/**
 * @deprecated 旧版的 Format 对象，为了向后兼容而保留。
 *   请直接导入 `NumberFormat` 和 `NumberFormatProvider` 组件。
 * @example
 * ```diff
 * - import { Format } from '@your-package/number-format';
 * + import { NumberFormat, NumberFormatProvider } from '@your-package/number-format';
 *
 * - <Format.Provider>
 * + <NumberFormatProvider>
 *     ...
 * -   <Format.Number value={123} />
 * +   <NumberFormat value={123} />
 *     ...
 * - </Format.Provider>
 * + </NumberFormatProvider>
 * ```
 */
export const Format = {
  Number: NumberFormat,
  Provider: NumberFormatProvider,
};
