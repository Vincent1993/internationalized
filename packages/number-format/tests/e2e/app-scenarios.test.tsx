import { render, screen } from '@testing-library/react';
import React from 'react';
import { NumberFormat, AutoMetricNumber } from '../../src/components';
import { NumberFormatProvider } from '../../src/core';
import type { MetricFormatRule } from '../../src/core/types';

describe('集成测试', () => {
  describe('完整的格式化流程', () => {
    it('应该支持从基础数字到复杂格式化的完整流程', () => {
      const TestApp = () => (
        <NumberFormatProvider
          options={{
            locale: 'en-US',
            useGrouping: true,
          }}
        >
          <div>
            {/* 基础数字格式化 */}
            <NumberFormat
              value={1234567.89}
              options={{ style: 'decimal', maximumFractionDigits: 2 }}
              data-testid='basic-decimal'
            />

            {/* 货币格式化 */}
            <NumberFormat
              value={1234.56}
              options={{ style: 'currency', currency: 'USD' }}
              data-testid='currency'
            />

            {/* 百分比格式化 */}
            <NumberFormat
              value={0.1234}
              options={{ style: 'percent', maximumFractionDigits: 1 }}
              data-testid='percent'
            />

            {/* 单位格式化 */}
            <NumberFormat
              value={123}
              options={{ style: 'unit', unit: 'meter' }}
              data-testid='unit'
            />
          </div>
        </NumberFormatProvider>
      );

      render(<TestApp />);

      // 验证所有格式化结果
      expect(screen.getByTestId('basic-decimal')).toHaveTextContent('1,234,567.89');
      expect(screen.getByTestId('currency')).toHaveTextContent('$1,234.56');
      expect(screen.getByTestId('percent')).toHaveTextContent('12.3%');
      expect(screen.getByTestId('unit')).toHaveTextContent('123 m');
    });

    it('应该支持自动指标格式化完整流程', () => {
      const rules: MetricFormatRule[] = [
        {
          pattern: /conversion|rate|percentage/i,
          options: { style: 'percent', maximumFractionDigits: 2 },
        },
        {
          pattern: /revenue|price|cost|amount/i,
          options: { style: 'currency', currency: 'USD' },
        },
        {
          pattern: /count|users|views|clicks/i,
          options: { style: 'decimal', maximumFractionDigits: 0 },
        },
      ];

      const TestAutoMetrics = () => (
        <NumberFormatProvider
          options={{
            locale: 'en-US',
            rules,
          }}
        >
          <div>
            <AutoMetricNumber name='conversion_rate' value={0.1234} data-testid='auto-conversion' />

            <AutoMetricNumber name='total_revenue' value={12345.67} data-testid='auto-revenue' />

            <AutoMetricNumber name='user_count' value={1234.56} data-testid='auto-count' />

            <AutoMetricNumber
              name='unknown_metric'
              value={123.456}
              options={{ style: 'decimal', maximumFractionDigits: 2 }}
              data-testid='auto-unknown'
            />
          </div>
        </NumberFormatProvider>
      );

      render(<TestAutoMetrics />);

      expect(screen.getByTestId('auto-conversion')).toHaveTextContent('12.34%');
      expect(screen.getByTestId('auto-revenue')).toHaveTextContent('$12,345.67');
      expect(screen.getByTestId('auto-count')).toHaveTextContent('1,235');
      expect(screen.getByTestId('auto-unknown')).toHaveTextContent('123.456');
    });
  });

  describe('Context层级与优先级', () => {
    it('应该正确处理多层Context的配置优先级', () => {
      const TestNestedContext = () => (
        <NumberFormatProvider
          options={{
            locale: 'en-US',
            style: 'decimal',
            useGrouping: true,
            maximumFractionDigits: 2,
          }}
        >
          <NumberFormat value={1234.5678} data-testid='outer-context' />

          <NumberFormatProvider
            options={{
              locale: 'de-DE',
              style: 'currency',
              currency: 'EUR',
              maximumFractionDigits: 1,
            }}
          >
            <NumberFormat value={1234.5678} data-testid='inner-context' />

            <NumberFormat
              value={1234.5678}
              options={{
                style: 'decimal', // 组件props覆盖Context
                maximumFractionDigits: 3, // 组件props覆盖Context
              }}
              data-testid='override-inner'
            />
          </NumberFormatProvider>
        </NumberFormatProvider>
      );

      render(<TestNestedContext />);

      // 外层Context效果
      expect(screen.getByTestId('outer-context')).toHaveTextContent('1,234.57');

      // 内层Context效果（德语货币格式）
      const innerElement = screen.getByTestId('inner-context');
      expect(innerElement.textContent).toMatchSnapshot('nested-inner-context');

      // 组件props覆盖内层Context
      const overrideElement = screen.getByTestId('override-inner');
      expect(overrideElement.textContent).toMatchSnapshot('nested-override');
    });
  });

  describe('性能和大量数据处理', () => {
    it('应该能处理大量数字格式化', () => {
      const numbers = Array.from({ length: 100 }, (_, i) => i * 123.45);

      const TestManyNumbers = () => (
        <NumberFormatProvider
          options={{
            locale: 'en-US',
            style: 'decimal',
            maximumFractionDigits: 2,
          }}
        >
          <div>
            {numbers.map((num, index) => (
              <NumberFormat key={index} value={num} data-testid={`number-${index}`} />
            ))}
          </div>
        </NumberFormatProvider>
      );

      const startTime = performance.now();
      render(<TestManyNumbers />);
      const endTime = performance.now();

      // 性能测试：100个数字格式化应该在合理时间内完成
      expect(endTime - startTime).toBeLessThan(1000); // 1秒内

      // 抽查几个结果
      expect(screen.getByTestId('number-0')).toHaveTextContent('0');
      expect(screen.getByTestId('number-1')).toHaveTextContent('123.45');
      expect(screen.getByTestId('number-10')).toHaveTextContent('1,234.5');
    });
  });

  describe('完整功能快照测试', () => {
    it('应该为完整的应用场景生成快照', () => {
      const rules: MetricFormatRule[] = [
        {
          name: 'financial',
          pattern: /revenue|profit|cost/i,
          options: { style: 'currency', currency: 'USD', minimumFractionDigits: 2 },
        },
        {
          name: 'performance',
          pattern: /rate|ratio|percentage/i,
          options: { style: 'percent', maximumFractionDigits: 1 },
        },
      ];

      const FullFeaturedApp = () => (
        <NumberFormatProvider
          options={{
            locale: 'en-US',
            useGrouping: true,
            rules,
          }}
        >
          <div data-testid='full-app'>
            <h1>Financial Dashboard</h1>

            <section>
              <h2>Revenue Metrics</h2>
              <AutoMetricNumber name='total_revenue' value={1234567.89} />
              <AutoMetricNumber name='operating_cost' value={456789.12} />
            </section>

            <section>
              <h2>Performance Metrics</h2>
              <AutoMetricNumber name='conversion_rate' value={0.1234} />
              <AutoMetricNumber name='growth_rate' value={0.0567} />
            </section>

            <section>
              <h2>Custom Formats</h2>
              <NumberFormat
                value={123456789}
                options={{
                  style: 'decimal',
                  notation: 'compact',
                  compactDisplay: 'short',
                }}
              />
              <NumberFormat
                value={0.123}
                options={{ style: 'per-mille' as any, maximumFractionDigits: 1 }}
              />
            </section>
          </div>
        </NumberFormatProvider>
      );

      const { container } = render(<FullFeaturedApp />);
      expect(container.firstChild).toMatchSnapshot('full-featured-app');
    });
  });
});
