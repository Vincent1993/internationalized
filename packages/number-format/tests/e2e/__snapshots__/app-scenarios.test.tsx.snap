// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`集成测试 > Context层级与优先级 > 应该正确处理多层Context的配置优先级 > nested-inner-context 1`] = `"1.234,6 €"`;

exports[`集成测试 > Context层级与优先级 > 应该正确处理多层Context的配置优先级 > nested-override 1`] = `"1.234,568"`;

exports[`集成测试 > 完整功能快照测试 > 应该为完整的应用场景生成快照 > full-featured-app 1`] = `
<div
  data-testid="full-app"
>
  <h1>
    Financial Dashboard
  </h1>
  <section>
    <h2>
      Revenue Metrics
    </h2>
    <span>
      1,234,567.89
    </span>
    <span>
      456,789.12
    </span>
  </section>
  <section>
    <h2>
      Performance Metrics
    </h2>
    <span>
      0.123
    </span>
    <span>
      0.057
    </span>
  </section>
  <section>
    <h2>
      Custom Formats
    </h2>
    <span>
      NaN
    </span>
    <span>
      123‰
    </span>
  </section>
</div>
`;
